File: block.py

import hashlib
import time

class Block:
    def __init__(self, transaction: dict, nonce: int, prev_hash: str = "0"):
        self.transaction = transaction
        self.nonce = nonce 
        self.timestamp = time.time()
        self.prev_hash = prev_hash
        self.hash = ""

    def compute_hash(self):
        block_bytes = str(self.__dict__).encode()
        block_hash = hashlib.sha256(block_bytes).hexdigest()
        return block_hash

File: app/app/lib/model/block.dart
class Block {
  late String from;
  late String to;
  late double amount;
  late int nonce;
  late double timestamp;
  late String hash;
  late String prevHash;
  late String title;
  late int id;

  Block({
    this.from = '',
    this.to = '',
    this.amount = 0,
    this.nonce = 0,
    this.timestamp = 0,
    this.hash = '',
    this.prevHash = '',
    this.title = 'Transaction',
    this.id = 0,
  });
}
